packages/security/src/K8sCredentialManager.ts

```diff
@@ -0,0 +1,229 @@
+                "data": {
+                    "credentials": credentials
+                }
+            }, "true");
+            Logger.getImperativeLogger().debug(`Successfully stored credentials as a kubernetes secret on namespace ${this.kubeConfig.namespace}`);
+        } catch (err) {
+            throw new ImperativeError({
+                msg: `Error when saving k8s secret ${this.secretName}`,
+                additionalDetails: err
+            });
+        }
+    }
+
+    /**
+     * Calls the K8s API to delete secrets with {@link K8sCredentialManager#service} and the
+     * account is passed to the function by Imperative.
+     *
+     * @param {string} account The account to set credentials
+     *
+     * @returns {Promise<void>} A promise that the function has completed.
+     *
+     * @throws {@link ImperativeError} if call to K8s API fails.
+     */
+    protected async deleteCredentials(account: string): Promise<void> {
+        try {
+            await this.readNamespacedSecret();
+            Logger.getImperativeLogger().debug(`Deleting k8s secret ${this.secretName}`);
+            await this.kc.deleteNamespacedSecret(this.secretName, this.kubeConfig.namespace, "true");
+        } catch (err) {
+            throw new ImperativeError({
+                msg: `Failed to delete secret ${this.secretName} in namespace ${this.kubeConfig.namespace}`,
+                additionalDetails: err.message
+            });
+        }
+    }
+
+    private get defaultService(): string {
+        return this.allServices[0];
+    }
+
+    private getMissingEntryMessage(account: string) {
+        let listOfServices = `  Service = `;
+        for (const service of this.allServices) {
+            listOfServices += `${service}, `;
+        }
+        const commaAndSpace = 2;
+        listOfServices = listOfServices.slice(0, -1 * commaAndSpace) + `\n  Account = ${account}\n\n`;
+
+        return "Could not find an entry in the credential vault for the following:\n" +
+            listOfServices +
+            "Possible Causes:\n" +
+            "  This could have been caused by any manual removal of credentials from your vault.\n\n" +
+            "Resolutions: \n" +
+            "  Recreate the credentials in the vault for the particular service in the vault.\n";
+    }
+
+    /**
+     * Retrieve data from kube config and return a {@link KubeConfig} object.
+     *
+     * @returns {k8s.CoreV1Api} the object allowing access to k8s api calls
+     *
+     * @throws {@link ImperativeError} if kube config file was not able to be opened.
+     */
+    private setupKubeConfig(): any {
+        try {
+            const kc: any = new k8s.KubeConfig();
+            kc.loadFromDefault();
+
+            // Check if login was performed
+            const currentContextNamespace = kc.getContextObject(kc.getCurrentContext())?.namespace;
+            const currentUser = kc.getCurrentUser();
+            if(!currentContextNamespace || !currentUser) {
```

Comment:
We should differentiate between not having a context and not being logged in.
Plus we should probably fallback to the cluster name instead of requiring a namespace since the `@kubernetes/client-node` package could return `undefined`.

```typescript
 // Check if login was performed
            const currentContext = kc.getContextObject(kc.getCurrentContext());
            if(!currentContext) {
                throw new Error("Current context was not found");
            }
            const currentUser = kc.getCurrentUser();
            if(!currentUser) {
                throw new Error("Current user was not found");
            }
            const k8sName = currentContext.name;
            const k8sNamespace = currentContext.namespace;
```

Reply:
this sounds like a good idea, as for the `k8sName` variable, perhaps we could fallback to it if the namespace is not found, a namespace should always be required/found (even if default but not recommended) once a login is performed into a kubernetes cluster.